rholang/src/test/scala/coop/rchain/rholang/interpreter/matcher/MatcherMonadSpec.scala

```diff
@@ -26,14 +38,22 @@ class MatcherMonadSpec extends FlatSpec {
     val possibleResults = Stream((0, 1), (0, 2))
     val computation     = Alternative[F].unite(possibleResults.pure[F])
     val sum             = computation.map { case (x, y) => x + y } <* charge[F](Cost(1))
-    val (cost, _)       = sum.runWithCost(Cost(possibleResults.size)).right.get
-    assert(cost.value == 0)
+    val t1 = (for {
+      _        <- cost.set(Cost(possibleResults.size))
+      _        <- runMatcher(sum)
+      phloLeft <- cost.get
+    } yield (assert(phloLeft.value == 0)))
```

Comment:
There's quite a lot of boilerplate happening in here, not to mention the impure `assert in map` and the possibility to accidentally not executing a computation by not calling 'run'.

Please extract a:
```scala
def runWithCost(f, phlo) = (for {
  _        <- cost.set(Cost(possibleResults.size))	
  result   <- f
  phloLeft <- cost.get
} yield (phloLeft, result)).unsafeRunSync
```
method, which will allow for retaining the original readability of the tests.

Reply:
An `.attempt` might be needed to retain the original signature expected by the tests - if so, let's have it.

Reply:
So be it!